test/fs.unittest.js

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+const { lstatReadlinkAbsolute } = require("../lib/util/fs");
+
+jest.mock("fs");
+
+describe("lstatReadlinkAbsolute", () => {
+	it("should call the callback with an error if fs.readlink fails", done => {
+		// We will mock fs.readlink to always return this error
+		const mockError = new Error("Mocked error");
+
+		// Mock fs.lstat to always return a symbolic link
+		jest.spyOn(fs, "lstat").mockImplementation((_, callback) => {
+			callback(null, { isSymbolicLink: () => true });
+		});
+
+		// Mock fs.readlink to always return the error above
+		jest.spyOn(fs, "readlink").mockImplementation((_, callback) => {
+			callback(mockError);
+		});
+
+		const callback = (err, result) => {
+			try {
+				expect(err).toBe(mockError);
+				expect(result).toBeUndefined();
+				done();
+			} catch (err_) {
+				done(err_);
+			}
+		};
+
+		lstatReadlinkAbsolute(fs, path.resolve("/some/path"), callback);
+	});
+});
```

Comment:
Let's use this - https://github.com/webpack/webpack/blob/main/test/FileSystemInfo.unittest.js for testing, because we use `lstatReadlinkAbsolute` there

Reply:
Ah, good idea, done! I moved the old test almost verbatim to the existing unit test, with one small addition: I reset the mocks on `fs.lstat` and `fs.readlink` at the end of the test to ensure they don't interfere with any future tests added to the file.

Let me know what you think, and if I should squash. Sorry for how long this took, I've been traveling.